src/containers/paint-editor.jsx

```diff
@@ -145,12 +170,58 @@ class PaintEditor extends React.Component {
     handleZoomReset () {
         resetZoom();
     }
+    setPaintEditor (paintEditor) {
+        this.paintEditor = paintEditor;
+    }
+    onMouseDown () {
+        if (this.props.isEyeDropping) {
+            const colorString = this.eyeDropper.colorString;
+            const callback = this.props.changeColorToEyeDropper;
+            
+            this.props.onDeactivateEyeDropper(this.props.previousMode);
+            this.stopEyeDroppingLoop();
+            if (!this.eyeDropper.hideLoupe) {
+                // If not hide loupe, that means the click is inside the canvas,
+                // so apply the new color
+                callback(colorString);
+            }
+            this.setState({colorInfo: null});
+        }
+    }
+    startEyeDroppingLoop () {
```

Comment:
paint-editor.jsx is getting kind of huge. I wonder if there's a good way to split code like this into other components or helper files before it gets unmanageable.

Reply:
I agree that it's getting huge, and that it'd be good to split things up. The reason I had this be in here rather than elsewhere is mostly to handle the context of the `Loupe` component. This component needs to exist at the context of the paint editor – rather than at the context of the canvas, for example – because we defined the behavior of it as hiding/going away on hover/click outside of the canvas itself. 

Given this issue, it seemed necessary to me for the component to exist at the paint editor level rather than in some separate component. It probably can be moved out, though, if that's a dealbreaker.

If we're looking to make the paint editor component smaller, though, I think we could get some easier refactors out of breaking out `Edit Tools` and `Mode Tools` into their own components.